bids/variables/variables.py

```diff
@@ -448,12 +448,31 @@ def resample(self, sampling_rate, inplace=False, kind='linear'):
         self.index = self._build_entity_index(self.run_info, sampling_rate)
 
         x = np.arange(n)
-        num = int(np.ceil(n * sampling_rate / old_sr))
+        # In Index we trust!
+        num = len(self.index)
+        # _build_entity_index computation above does it per run,
+        # which possibly provides a more stable solution and yadadada
+        num_computed = int(np.ceil(n * sampling_rate / old_sr))
+        num_diff = abs(num - num_computed)
+        if num_diff > 1:
+            raise RuntimeError(
+                "Our internal assumptions about resampling are too faulty "
+                "to continue.")
+        elif num_diff:
+            assert num_diff == 1  # the only way to get here
+            import warnings
+            warnings.warn(
+                "Probably due to multiple runs we ran into a slight "
+                "divergence between obtained number of samples in the index (computed "
+                "per each run) and overall estimate down/up-sampled samples. "
+                "But that is ok")
```

Comment:
we looked at the code more, and our verdict is that we just need to get rid of num_computed here altogether and use the length of the index, estimated (by going through runs and using their actual duration which is not a subject of up/down sampling effects of "ints").

The reason is -- the `n` is not "reliable". It is already a result of applying `ceil` operation in `to_dense` function (called by the test first to upsample by 10). so there is really no reliable way to get back the original number of points.

But also what is important probably is that `interp1d` below should also be applied per run, similarly to how to_dense is doing it I guess, to not leak values across the run boundaries.... but that is a separate issue